mlrun/model_monitoring/controller.py

```diff
@@ -265,7 +278,40 @@ def _should_monitor_endpoint(endpoint: mlrun.common.schemas.ModelEndpoint) -> bo
             # Is the model endpoint not a router endpoint? Router endpoint has no feature stats
             and endpoint.metadata.endpoint_type.value
             != mm_constants.EndpointType.ROUTER.value
-        )
+        ):
+            with _BatchWindowGenerator(
+                project=endpoint.metadata.project,
+                endpoint_id=endpoint.metadata.uid,
+            ) as batch_window_generator:
+                if application_names != batch_window_generator.get_application_list():
+                    return True
+                elif (
+                    not batch_window_generator.get_min_last_analyzed()
+                    or batch_window_generator.get_min_last_analyzed()
+                    <= int(endpoint.status.last_request.timestamp())
+                ):
+                    return True
+                else:
+                    logger.info(
+                        "All the possible intervals were already analyzed, didn't push regular event",
+                        endpoint_id=endpoint.metadata.uid,
+                        last_analyzed=datetime.datetime.fromtimestamp(
+                            batch_window_generator.get_min_last_analyzed(),
+                            tz=datetime.timezone.utc,
+                        ),
+                        last_request=endpoint.status.last_request,
+                    )
+        else:
+            logger.info(
```

Comment:
down the road we should consider if this is not spamming the controller logs too much